zebra-consensus/src/transaction/tests.rs

```diff
@@ -3446,87 +3373,65 @@ fn fill_action_with_note_encryption_test_vector(
+            let mut transaction = v5_transactions(net.block_iter())
+                .find(|tx| tx.is_coinbase())
+                .expect("coinbase V5 tx");
+
+            let shielded_data = insert_fake_orchard_shielded_data(&mut transaction);
+            shielded_data.flags = Flags::ENABLE_SPENDS | Flags::ENABLE_OUTPUTS;
+
+            let action =
+                fill_action_with_note_encryption_test_vector(&shielded_data.actions[0].action, v);
+            let sig = shielded_data.actions[0].spend_auth_sig;
+            shielded_data.actions = vec![AuthorizedAction::from_parts(action, sig)]
+                .try_into()
+                .unwrap();
 
-        assert_eq!(
-            check::coinbase_outputs_are_decryptable(
-                &transaction,
-                &network,
-                NetworkUpgrade::Nu5.activation_height(&network).unwrap(),
-            ),
-            Ok(())
-        );
+            assert_eq!(
+                check::coinbase_outputs_are_decryptable(
+                    &transaction,
+                    &net,
+                    NetworkUpgrade::Nu5.activation_height(&net).unwrap(),
+                ),
+                Ok(())
+            );
+        }
     }
 }
 
-/// Test if random shielded outputs are NOT decryptable with an all-zero outgoing
-/// viewing key.
+/// Test if random shielded outputs are NOT decryptable with an all-zero outgoing viewing key.
 #[test]
 fn shielded_outputs_are_not_decryptable_for_fake_v5_blocks() {
-    let network = Network::new_default_testnet();
-
     for v in zebra_test::vectors::ORCHARD_NOTE_ENCRYPTION_VECTOR.iter() {
-        // Find a transaction with no inputs or outputs to use as base
-        let mut transaction =
-            fake_v5_transactions_for_network(&network, zebra_test::vectors::TESTNET_BLOCKS.iter())
-                .rev()
-                .find(|transaction| {
-                    transaction.inputs().is_empty()
-                        && transaction.outputs().is_empty()
-                        && transaction.sapling_spends_per_anchor().next().is_none()
-                        && transaction.sapling_outputs().next().is_none()
-                        && transaction.joinsplit_count() == 0
-                })
-                .expect("At least one fake V5 transaction with no inputs and no outputs");
-
-        let shielded_data = insert_fake_orchard_shielded_data(&mut transaction);
-        shielded_data.flags = zebra_chain::orchard::Flags::ENABLE_SPENDS
-            | zebra_chain::orchard::Flags::ENABLE_OUTPUTS;
-
-        let action =
-            fill_action_with_note_encryption_test_vector(&shielded_data.actions[0].action, v);
-        let sig = shielded_data.actions[0].spend_auth_sig;
-        shielded_data.actions = vec![AuthorizedAction::from_parts(action, sig)]
-            .try_into()
-            .unwrap();
+        for net in Network::iter() {
+            let mut tx = v5_transactions(zebra_test::vectors::TESTNET_BLOCKS.iter())
```

Comment:
Nice catch updating this test to use a coinbase transaction!

Did you mean to use `block_iter()` here?

```suggestion
            let mut tx = v5_transactions(net.block_iter())
```

Related but wasn't changed in this PR: Returning early from `coinbase_tx_no_prevout_joinsplit_spend()` but not `coinbase_outputs_are_decryptable()` if `!tx.is_coinbase()` seems inconsistent (albeit unimportant), we may want to update one of them.

Reply:
Thanks for spotting the inconsistency. I'll open another PR.

Reply:
Or I do it in this PR since it's a minor clean-up, even though it's not related to this PR.

Reply:
> Related but wasn't changed in this PR: Returning early from `coinbase_tx_no_prevout_joinsplit_spend()` but not `coinbase_outputs_are_decryptable()` if `!tx.is_coinbase()` seems inconsistent (albeit unimportant), we may want to update one of them.

Done in [6928e06](https://github.com/ZcashFoundation/zebra/pull/9098/commits/6928e06d52c18776cec66495944f4f4fa28c722d).